### PR TITLE
G10: Post `docker pull` commands for local testing

### PR DESCRIPTION
**Pull request by dsotirakis**
**What is this feature?**

Makes it able for devs to test their changes per commit, using a docker image. The docker image will appear as a check run after the `publish-images` step has successfully ran.

**Why do we need this feature?**

As part of the testing phase before Grafana 10

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-delivery/issues/121

### CODE REVIEW

**Review by zerok — APPROVED** [review]

**Comment by dsotirakis** [comment]
Use the following command to pull the docker image:

`docker pull grafana/grafana:test`

**Review by zerok — CHANGES_REQUESTED** [review]
We might need a way to restrict building these images further so that only specific PRs would do that.

**Comment by dsotirakis** [comment]
> We might need a way to restrict building these images further so that only specific PRs would do that.

@zerok discussed this already - we will use a GitHub app which restricts that to org members only :)

File: scripts/drone/vault.star [hunk 1]
```diff
@@ -123,6 +123,16 @@ def secrets():
             "infra/data/ci/grafana-release-eng/enterprise2",
             "security_prefix",
         ),
+        vault_secret(
+            "grafana_upstream_repo_main",
+            "infra/data/ci/grafana-release-eng/grafana-enterprise",
+            "upstream-repo-main",
+        ),
+        vault_secret(
```
Comment (zerok) [inline scripts/drone/vault.star:131]: This one doesn't seem to be used anywhere :)

Comment (dsotirakis) [reply scripts/drone/vault.star]: Oops, massive c/p :D Thanks!

**Review by zerok — APPROVED** [review]

**Pull request merged**